helix-core/src/main/java/org/apache/helix/monitoring/metrics/model/Metric.java

```diff
@@ -24,7 +24,7 @@
 /**
  * Defines a generic metric interface.
  */
-public interface Metric {
+public interface Metric<T> {
```

Comment:
Let's comment the T is the input data, not necessary to be the output type of the metric object.
For example, Histogram based metrics emits float or double.